front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -209,46 +214,38 @@ export default function AssistantBuilder({
   };
 
   const formValidation = useCallback(async () => {
-    let valid = true;
+    const modelConfig = SUPPORTED_MODEL_CONFIGS.filter(
+      (config) =>
+        config.modelId === builderState.generationSettings.modelSettings.modelId
+    )[0];
+    if (!modelConfig) {
+      // unreachable
+      throw new Error("Model configuration not found");
+    }
 
-    const { handleValid, handleErrorMessage } = await validateHandle({
+    const { handleErrorMessage } = await validateHandle({
       owner,
       handle: builderState.handle,
       initialHandle: initialBuilderState?.handle,
       checkUsernameTimeout,
     });
-    valid = handleValid;
     setAssistantHandleError(handleErrorMessage);
 
-    if (
-      !builderState.description?.trim() ||
-      !builderState.instructions?.trim()
-    ) {
-      valid = false;
+    let localDescriptionError: string | null = null;
```

Comment:
I think all the "local state" makes this validation function hard to read. 
We're creating a bunch of new vars that we don't really need to have.

Here's what IMHO is easier to read: 


```js
  const formValidation = useCallback(async () => {
    const modelConfig = SUPPORTED_MODEL_CONFIGS.filter(
      (config) =>
        config.modelId === builderState.generationSettings.modelSettings.modelId
    )[0];
    if (!modelConfig) {
      // unreachable
      throw new Error("Model configuration not found");
    }

    let isValid = true;

    // Validate Assistant handle
    const { handleErrorMessage } = await validateHandle({
      owner,
      handle: builderState.handle,
      initialHandle: initialBuilderState?.handle,
      checkUsernameTimeout,
    });
    if (handleErrorMessage) {
      setAssistantHandleError(handleErrorMessage);
      isValid = false;
    }

    // Validate Assistant description
    if (!builderState.description?.trim()) {
      setDescriptionError("You must provide a description.");
      isValid = false;
    }

    // Validate Assistant instructions
    let localInstructionError: string | null = null;
    if (!builderState.instructions?.trim()) {
      localInstructionError = "You must provide some instructions.";
    } else if (
      builderState.instructions &&
      builderState.instructions.length > INSTRUCTIONS_MAXIMUM_CHARACTER_COUNT
    ) {
      localInstructionError = `Instructions must be less than ${INSTRUCTIONS_MAXIMUM_CHARACTER_COUNT} characters.`;
    } else if (
      builderState.instructions &&
      builderState.instructions.trim().length / 4 >
        modelConfig.contextSize * 0.9
    ) {
      localInstructionError = `Instructions may exceed context size window.`;
    }
    if (localInstructionError) {
      setInstructionsError(localInstructionError);
      isValid = false;
    }

    // Validate assistant actions
    const actionErrors = removeNulls(
      builderState.actions.map((a) => hasActionError(a))
    );
    if (actionErrors.length) {
      isValid = false;
      setActionsErrors(actionErrors);
    }

    setSubmitEnabled(isValid);
  }, [builderState, owner, initialBuilderState?.handle]);
```

Reply:
> I think all the "local state" makes this validation function hard to read.
> We're creating a bunch of new vars that we don't really need to have.

> Here's what IMHO is easier to read:

You are suggesting to back to the previous state (more or less).
I didn't go down that road because this duplicate the truth : valid is "you have no error" but now your truth is a flag that you have to maintain separately.

This lead to problem such as the issue that lead me to do the PR where missing instructions or descriptions would flag as invalid but without any user feedback.

That being said, that made me realize that to be 100% coherent with what I just said, i should remove the valid state altogether (i'll).